File: .NET & JAVA/JAVA/VehicleConfigurator/src/main/java/com/example/entities/AlternateComponent.java
package com.example.entities;

import java.math.BigDecimal;

import com.fasterxml.jackson.annotation.JsonIgnore;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;

@Entity
@Table(name = "alternate_component_master")
public class AlternateComponent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int altId;

    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "model_id")  
    private Model model;
    
    @ManyToOne
    @JsonIgnore
    @JoinColumn(name = "comp_id")  
    private Component baseComponent;

    @ManyToOne
    @JoinColumn(name = "alt_comp_id") 
    private Component alternateComponent;

    private BigDecimal deltaPrice;

    public int getAltId() {
		return altId;
	}

	public void setAltId(int altId) {
		this.altId = altId;
	}

	public Model getModel() {
		return model;
	}

	public void setModel(Model model) {
		this.model = model;
	}

	public Component getBaseComponent() {
		return baseComponent;
	}

	public void setBaseComponent(Component baseComponent) {
		this.baseComponent = baseComponent;
	}

	public Component getAlternateComponent() {
		return alternateComponent;
	}

	public void setAlternateComponent(Component alternateComponent) {
		this.alternateComponent = alternateComponent;
	}

	public BigDecimal getDeltaPrice() {
		return deltaPrice;
	}

	public void setDeltaPrice(BigDecimal deltaPrice) {
		this.deltaPrice = deltaPrice;
	}
    
}

File: DotNet/Models/AlternateComponent.cs
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace Vehicle_Configurator_Project.Models
{
    [Table("alternate_component_master")]
    public class AlternateComponent
    {
        [Key]
        [Column("alt_id")]
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        public int AltId { get; set; }

        [ForeignKey(nameof(Model))]
        [Column("model_id")]
        [JsonIgnore]
        public int ModelId { get; set; }
        public Model Model { get; set; }

        [ForeignKey(nameof(BaseComponent))]
        [Column("comp_id")]
        [JsonIgnore]
        public int BaseComponentId { get; set; }
        public Component BaseComponent { get; set; }

        [ForeignKey(nameof(AlternateComponentEntity))]
        [Column("alt_comp_id")]
        public int AlternateComponentId { get; set; }
        public Component AlternateComponentEntity { get; set; }

        [Column("delta_price")]
        public decimal? DeltaPrice { get; set; }
    }
}
